### PR TITLE
Custom domains: hide https option

### PR DESCRIPTION
**Pull request by stsewd**
We only show the option to people how don't have https set to true in their custom domain, so they can migrate.

Hopefully one day we can migrate all domains to https.

Closes https://github.com/readthedocs/readthedocs.org/issues/9469

### CODE REVIEW

File: readthedocs/projects/forms.py [hunk 1]
```diff
@@ -706,10 +706,15 @@ def __init__(self, *args, **kwargs):
...
-        instance = getattr(self, 'instance', None)
-        if instance and instance.pk:
+        if self.instance.pk:
             self.fields['domain'].disabled = True
 
+        # Remove the https option at creation,
+        # but show it if the domain is already marked as http only,
+        # so users can upgrade.
+        if not self.instance.pk or self.instance.https:
+            self.fields.pop("https")
```
Comment (benjaoming) [inline readthedocs/projects/forms.py:716]: Never allowing downgrades seems fine, not sure if it's necessary.. but I can see the security issue, so I'm good with it :+1: Always possible to ask support in a very very very rare case.

File: readthedocs/projects/forms.py [hunk 2]
```diff
@@ -706,10 +706,15 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
         # Disable domain manipulation on Update, but allow on Create
-        instance = getattr(self, 'instance', None)
-        if instance and instance.pk:
+        if self.instance.pk:
```
Comment (benjaoming) [inline readthedocs/projects/forms.py:709]: I hope this works :) I don't remember if there can be instance=None on a creation form? But I guess you've checked that one :)

**Review by benjaoming — APPROVED** [review]
LGTM! Haven't tested locally, but I think I can trust that you've tried both creation and update and that's sufficient :+1:

File: readthedocs/projects/forms.py (hunk 2 above)
Comment (stsewd) [reply readthedocs/projects/forms.py:709]: This attribute was always defined in my local testing, I think it's None if explicitly set to None :man_shrugging:

**Comment by stsewd** [comment]
Just pushed a small change to delete the class override, since we don't need it anymore now that we are matching the same behavior from .com.

**Pull request merged**